packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -721,61 +779,133 @@ describe('internal-plugin-metrics', () => {
+      });
+
+      const checkLocusError = (payload: any, isExpectedToBeCalled: boolean) => {
+        const res = cd.generateClientEventErrorPayload(payload);
+        const expectedResult = {
           category: 'signaling',
           errorDescription: 'NewLocusError',
           fatal: true,
           name: 'other',
           shownToUser: false,
           serviceErrorCode: 2400000,
           errorCode: 4008,
-        });
+        };
+
+        if (isExpectedToBeCalled) {
+          assert.deepEqual(res, expectedResult);
+        } else {
+          assert.notDeepEqual(res, expectedResult);
+        }
+      }
+
+      it('should return default new locus event error payload correctly if locus error is recognized', () => {
+        checkLocusError({body: {errorCode: 2400000}}, true);
       });
 
-      it('should generate event error payload correctly for nested error', () => {
-        const res = cd.generateClientEventErrorPayload({error: {body: {errorCode: 2409005}}});
-        assert.deepEqual(res, {
-          category: 'expected',
-          errorDescription: 'StartRecordingFailed',
-          fatal: true,
-          name: 'other',
-          shownToUser: false,
-          errorCode: 4029,
-          serviceErrorCode: 2409005,
-        });
+      it('should not return default new locus event error payload correctly if locus is not recognized', () => {
+        checkLocusError({body: {errorCode: 1400000}}, false);
       });
 
-      it('should return default meeting info lookup error payload correctly if not locus error', () => {
-        const res = cd.generateClientEventErrorPayload({body: {errorCode: 9400000}});
-        assert.deepEqual(res, {
+      const checkMeetingInfoError = (payload: any, isExpectedToBeCalled: boolean) => {
+        const res = cd.generateClientEventErrorPayload(payload);
+        const expectedResult = {
           category: 'signaling',
           errorDescription: 'MeetingInfoLookupError',
           fatal: true,
           name: 'other',
           shownToUser: false,
-          serviceErrorCode: 9400000,
+          serviceErrorCode: undefined,
           errorCode: 4100,
-        });
+        };
+
+        if (isExpectedToBeCalled) {
+          assert.deepEqual(res, expectedResult);
+        } else {
+          assert.notDeepEqual(res, expectedResult);
+        }
+      }
+
+      it('should return default meeting info lookup error payload if data.meetingInfo was found on error body', () => {
+        checkMeetingInfoError({body: {data: {meetingInfo: 'something'}}}, true);
       });
 
-      it('should return undefined if no error code provided', () => {
-        const res = cd.generateClientEventErrorPayload({body: {errorCode: undefined}});
-        assert.deepEqual(res, undefined);
+      it('should return default meeting info lookup error payload if body.url contains wbxappapi', () => {
+        checkMeetingInfoError({body: {url: '1234567-wbxappapiabcdefg'}}, true);
+      });
+
```

Comment:
should we also have a test case "should return default meeting info lookup error payload if the error code is one of WbxAppAPI ones? or do we not care about the actual error code and only rely on data.meetingInfo or the url containing "wbxappapi" string? (this comment is related to my previous one [here](https://github.com/webex/webex-js-sdk/pull/3074/files#r1334594631))

Reply:
The case in which the error code is one of the WbxAppAPI ones is covered in tests in line 823-837. 
<img width="776" alt="image" src="https://github.com/webex/webex-js-sdk/assets/56999622/4eaaf5cc-6024-4bdd-8600-348e15672491">

Although the test is using code 2409005 which is a Locus code, it applies the same for WbxAppAPI bc these known codes are written in the same SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP.  

If this map does not capture the error, then we at least try to see if we can identify it as a WbxAppAPI Meeting Info error and catalogue it as a general MEETING_INFO_LOOKUP_ERROR_CLIENT_CODE (error 4100).  That is what tests in line 885 - 896 are about.